### ISSUE TITLE
Greeting by first name on dashboard

### ISSUE THREAD

**Issue by mbtools** [exercism/exercism#6277]
Currently, you are greeted by you user name... not very nice

![image](https://user-images.githubusercontent.com/59966492/157664454-1650253d-f598-4105-93d8-436792bf97b9.png)

I suggest to change it to your first name:

**Welcome back, Marc!**

**Comment by kotp** [comment]
> not very nice

The user name was (presumably) chosen by the user, some users consider it a friendly nickname.  I am not sure we can assume that it is not very nice.  But I do agree it would be nice to be able to select which way you would want to be addressed.  When addressed by my user name, I smile, because it has 20+ years of memories behind it.  When I see my first name, I assume I am being greeted by someone with a different set of memories attached.

I do agree that we could have a choice of how to be greeted though.

Perhaps an addition in [settings] where we can choose how to be addressed in other ways, it should include direct reference by name as well?

[settings]: https://exercism.org/settings

**Comment by iHiD** [comment]
@mbtools The main challenge here is that we don't have first names stored, only full names. And it's impossible to reliably extract a first name from a full name. 

So it's not something we can just automatically do, but we could add an "how do you want to be addressed" field, which could be used here, and also in mentoring, etc (as @kotp suggests)

**Comment by NobbZ** [comment]
@mbtools Some consider it "not very nice", other though consider it the only true way to address them.

Even though I am very open with my given/public/RL name, I simply prefer to be addressed as "NobbZ" in most circumstances.

In my case it is already at a stage that even my parents, my wife, my kids, my colleuges, and business partners and customers call me that way. It feels unnatural to me to read/hear my actual name.

Though I can understand that some would prefer to be addressed by their first name, others by their family name + "Mr/Mrs", "Dr" or whatever.

Though using the nickname as primary handle for people is common in the web. I think this is why this form has been choosen initially, and I think we can make exercism a friendlier place by asking the user about **their** prefered form, as we already ask for pronouns.

**Comment by mbtools** [comment]
I apologize. I temporarily forgot that it's a site for developers 😉 

Well, you already have a setting for "How would you like to be addressed?" but it's just about pronouns...

![image](https://user-images.githubusercontent.com/59966492/157868257-3f481b11-355f-4851-8306-b2b2f4b0bd81.png)

The obvious solution is to let the user define how they would like to be greeted.

**Comment by SleeplessByte** [comment]
Yes, @mbtools , I think that's exactly what @iHiD is proposing. I would like to see that too.

**Comment by SaschaMann** [comment]
Related discussion in https://github.com/exercism/exercism/issues/5778

**Comment by iHiD** [comment]
I think we need two:
1. Short one (for scenarios like the dashboard, and in sentences)
2. Long one (for scenarios like appear in mentor comments as per the discussion on https://github.com/exercism/exercism/issues/5778)

I wouldn't just want "Jeremy" for (2). And I wouldn't want "Jeremy Walker" for one.

**Comment by mbtools** [comment]
I guess you meant "closed/not planned" instead of "completed" or is there some change to the site that I'm missing?

**Comment by kotp** [comment]
Completed as it relates to this repository, but retention of the latest state as indicated by the May 10th 2022 status update above.

**Comment by SleeplessByte** [comment]
@mbtools it is still on the roadmap!
